src/main/scala/is/hail/check/Gen.scala

```diff
@@ -404,8 +406,12 @@ class Gen[+T](val gen: (Parameters) => T) extends AnyVal {
   // FIXME should be non-strict
   def withFilter(f: (T) => Boolean): Gen[T] = Gen { (p: Parameters) =>
     var x = apply(p)
-    while (!f(x))
+    var i = 0
+    while (!f(x)) {
+      assert(i < 100)
```

Comment:
It looks like you're doing rejection sampling that dies if unsuccessful after 100 attempts. Is the idea that our tests are seeded / deterministic so we need not worry about random failures?